src/llama.cpp

```diff
@@ -4371,9 +4500,20 @@ static void llm_load_hparams(
         GGML_ASSERT(hparams.n_expert_used == 0);
     }
 
+    // zero-out the per-layer hparams
+    std::fill(hparams.n_head_arr.begin(),    hparams.n_head_arr.end(),    0);
+    std::fill(hparams.n_head_kv_arr.begin(), hparams.n_head_kv_arr.end(), 0);
+    std::fill(hparams.n_ff_arr.begin(),      hparams.n_ff_arr.end(),      0);
+
+    ml.get_key_or_arr(LLM_KV_FEED_FORWARD_LENGTH,  hparams.n_ff_arr,   hparams.n_layer);
+    ml.get_key_or_arr(LLM_KV_ATTENTION_HEAD_COUNT, hparams.n_head_arr, hparams.n_layer);
+
+    GGML_ASSERT(hparams.n_head() > 0);
```

Comment:
@ggerganov (from your change in <https://github.com/ggerganov/llama.cpp/commit/29ab5a0ed121cdbcad3012e07154a6d3132e01a6>) This will fail for non-Transformer models like Mamba, and even partially non-Transformer models like Jamba.

(This was also the reason for the checks before divisions)

Reply:
Thank, I forgot about this. Will try to resolve